scripts/Interface/ui/sans_isis/beam_centre.py

```diff
@@ -0,0 +1,171 @@
+from __future__ import (absolute_import, division, print_function)
+
+from abc import ABCMeta, abstractmethod
+
+from PyQt4 import QtGui, QtCore
+from six import with_metaclass
+import ui_beam_centre
+from mantidqtpython import MantidQt
+
+try:
+    _fromUtf8 = QtCore.QString.fromUtf8
+except AttributeError:
+    def _fromUtf8(s):
+        return s
+
+
+class BeamCentre(QtGui.QWidget, ui_beam_centre.Ui_BeamCentre):
+    class BeamCentreListener(with_metaclass(ABCMeta, object)):
+        """
+        Defines the elements which a presenter can listen to for the beam centre finder
+        """
+        @abstractmethod
+        def on_run_clicked(self):
+            pass
+
+    def __init__(self, parent=None):
+        super(BeamCentre, self).__init__(parent)
+        self.setupUi(self)
+        self._setup_log_widget()
+
+        # Hook up signal and slots
+        self.connect_signals()
+        self._beam_centre_tab_listeners = []
+
+    def _setup_log_widget(self):
+        self.log_widget = MantidQt.MantidWidgets.MessageDisplay(self.groupBox_2)
+        self.log_widget.setMinimumSize(QtCore.QSize(491, 371))
+        self.log_widget.setObjectName(_fromUtf8("log_widget"))
+        self.gridLayout_4.addWidget(self.log_widget, 0, 1, 4, 1)
+        self.log_widget.setSource("CentreFinder")
+        self.log_widget.attachLoggingChannel()
+
+    def connect_signals(self):
+        self.run_button.clicked.connect(self.on_run_clicked)
+
+    def add_listener(self, listener):
+        if not isinstance(listener, BeamCentre.BeamCentreListener):
+            raise ValueError("The listener ist not of type BeamCentreListener but rather {}".format(type(listener)))
+        self._beam_centre_tab_listeners.append(listener)
+
+    def clear_listeners(self):
+        self._beam_centre_tab_listeners = []
+
+    def _call_beam_centre_tab_listeners(self, target):
+        for listener in self._beam_centre_tab_listeners:
+            target(listener)
+
+    def on_run_clicked(self):
+        self._call_beam_centre_tab_listeners(lambda listener: listener.on_run_clicked())
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # Actions
+    # ------------------------------------------------------------------------------------------------------------------
+    def set_options(self, options):
+        self.r_min = options.r_min
+        self.r_max = options.r_max
+        self.max_iterations = options.max_iterations
+        self.tolerance = options.tolerance
+        self.left_right = options.left_right
+        self.up_down = options.up_down
+
+    def update_simple_line_edit_field(self, line_edit, value):
+        if value:
```

Comment:
I can see why you have put the if here, but we should avoid if statements in the view. Is it possible to put the check in the presenter?